pkg/controller/jivavolume/jivavolume_controller.go

```diff
@@ -238,15 +238,13 @@ func (r *ReconcileJivaVolume) shouldReconcile(cr *jv.JivaVolume) (bool, error) {
 // 2. Create controller deploy
 // 3. Create replica statefulset
 func (r *ReconcileJivaVolume) bootstrapJiva(cr *jv.JivaVolume, reqLog logr.Logger) (err error) {
-	defer r.finally(err, cr, reqLog)
```

Comment:
Here the defer call was made before the error get populated and as it is not a pointer it always passes it as nil.